### PR TITLE
MGMT-9128: Remove V1 GetFreeAddresses (GET)

### PR DESCRIPTION
**Pull request by nmagnezi**
## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @filanov 
/cc @carbonin 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -5227,67 +5226,8 @@ func (b *bareMetalInventory) validateDNSDomain(cluster common.Cluster, params in
...
 func (b *bareMetalInventory) GetFreeAddresses(ctx context.Context, params installer.GetFreeAddressesParams) middleware.Responder {
-	log := logutil.FromContext(ctx, b.log)
-
-	results, err := b.getFreeAddresses(ctx, params, log)
-	if err != nil {
-		log.WithError(err).Warn("GetFreeAddresses")
-		return common.GenerateErrorResponder(err)
-	}
-	return installer.NewGetFreeAddressesOK().WithPayload(results)
+	return common.NewApiError(http.StatusNotImplemented, errors.New(APINotImplemented))
```
Comment (carbonin) [inline internal/bminventory/inventory.go:5230]: I feel like this should probably be a 404.
A 5xx error usually indicates a server problem that can't be resolved by the client, but this is a client problem.
Additionally what is happening in this case doesn't really match the description of 501 (ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501)

cc @avishayt what do you think?

Comment (filanov) [reply internal/bminventory/inventory.go:5230]: We did the same with the wrapper that you implemented so it make sense. @nmagnezi sorry if i confused you with that.

Comment (nmagnezi) [reply internal/bminventory/inventory.go:5230]: Ack. will change to 404.

**Review by filanov — APPROVED** [review]

**Pull request merged**